src/serialiser.rs

```diff
@@ -13,10 +13,11 @@ impl From<(Circuit, Vec<Fr>)> for CurveAcir {
         // so we extract all of the arithmetic gates only
         let circ = circ_val.0;
         let arith_gates: Vec<_> = circ
-            .gates
+            .opcodes
             .into_iter()
-            .filter(|gate| gate.is_arithmetic())
-            .map(|gate| CurveAcirArithGate::from(gate.arithmetic()))
+            .filter(|opcode| opcode.is_arithmetic())
+            // TODO: What is a better way to handle the Option returned by opcode.arithmetic()
+            .map(|opcode| CurveAcirArithGate::from(opcode.arithmetic().unwrap()))
```

Comment:
@vezenovm What is the condition in which `opcode.arithmetic()` returns `None`? Is it ensured to be `Some` if `is_arithmetic()` is true?

Reply:
Yes that is what it does. `arithmetic()` just returns the Expression stored inside of the arithmetic opcode. We technically are filtering out any not arithmetic gates before the map but if you are matching against an opcode to find `Opcode::Arithmetic` you have to have an option.